### ISSUE TITLE
minor ticks don't respect rcParam xtick.top / ytick.right

### ISSUE THREAD

**Issue by drammock** [matplotlib/matplotlib#6408]
**Expected behavior:** `plt.rc('xtick', top=False)` should prevent any ticks from displaying at the top edge of the axes.

**What happens instead:**  minor ticks are visible at the top edge.

```python
import numpy as np
import matplotlib.pyplot as plt
# set up plot style
plt.rc('xtick.minor', size=15)  # minor ticks extra long so they're obvious
plt.rc('xtick', top=False)      # should be no ticks at top (major or minor)
# plot data
data = np.random.rand(100).reshape(10, 10)
fig, ax = plt.subplots(1, 1)
ax.imshow(data)
# customize tick locations
ax.set_xticks(np.arange(10)[1::2], minor=False)  # this respects top=False
ax.set_xticks(np.arange(10)[::2], minor=True)  # this does not
```
![mwe](https://cloud.githubusercontent.com/assets/1810515/15201103/38e200ca-17a2-11e6-9e07-42d67889e14a.png)

matplotlib version: 1.5.1+1643.g0423430 compiled from source on Linux (Xubuntu 14.04 64-bit)

**Comment by tacaswell** [comment]
Good catch!  Thank you for the very clear example of what is going wrong. 

With the code in #6410 I get

![so](https://cloud.githubusercontent.com/assets/199813/15203315/53d7479a-17d0-11e6-8367-733cb2821280.png)

for your example code.

**Comment by drammock** [comment]
Wow, fast fix! Thanks @tacaswell

**Comment by efiring** [comment]
Closed by #6410
